test/unit/core/esriMapUtils.spec.js

```diff
@@ -0,0 +1,180 @@
+            };
+            esriMapUtils.addCustomBasemap('delorme', definition).then(function(esriBasemaps) {
+                expect(esriBasemaps.delorme).toEqual(definition);
+            });
+            $rootScope.$digest();
+        });
+
+        it('should not add definition w/o urls or baseMapLayers', function() {
+            var definition = {
+                title: 'DeLorme',
+                thumbnailUrl: 'http://servername.fqdn.suffix/images/thumbnail_2014-11-25_61051.png'
+            };
+            esriMapUtils.addCustomBasemap('delorme', definition).then(function(esriBasemaps) {
+                expect(esriBasemaps.delorme).toBe(undefined);
+            });
+            $rootScope.$digest();
+        });
+    });
+
+    describe('createMap', function() {
+        var Map;
+        var Extent;
+
+        beforeEach(function(){
+            Map = jasmine.createSpy();
+            Extent = jasmine.createSpy();
+            deferred.resolve([Map, Extent]);
+        });
+
+        it('should create a map with no options', function() {
+            esriMapUtils.createMap('notARealNodeId').then(function(/*map*/) {
+                expect(Map.calls.argsFor(0)).toEqual(['notARealNodeId', undefined]);
+                expect(Extent.calls.count()).toEqual(0);
+            });
+            $rootScope.$digest();
+        });
+
+        it('should create a map with options', function() {
+            var options = {
+                center: [-111.879655861, 40.571338776],
+                zoom: 13,
+                sliderStyle: 'small'
+            };
+            esriMapUtils.createMap('notARealNodeId', options).then(function(/*map*/) {
+                expect(Map.calls.argsFor(0)).toEqual(['notARealNodeId', options]);
+                expect(Extent.calls.count()).toEqual(0);
+            });
+            $rootScope.$digest();
+        });
+
+        it('should normalize extent option', function() {
+            var options = {
+                extent: {
+                    'xmin': -1556394,
+                    'ymin': -2051901,
+                    'xmax': 1224513,
+                    'ymax': 2130408,
+                    'spatialReference': {
+                        'wkid': 102003
+                    }
+                }
+            };
+            esriMapUtils.createMap('notARealNodeId', options).then(function(/*map*/) {
+                expect(Map.calls.argsFor(0)).toEqual(['notARealNodeId', options]);
+                expect(Extent.calls.argsFor(0)).toEqual([{
+                    'xmin': -1556394,
+                    'ymin': -2051901,
+                    'xmax': 1224513,
+                    'ymax': 2130408,
+                    'spatialReference': {
+                        'wkid': 102003
+                    }
+                }]);
+            });
+            $rootScope.$digest();
+        });
+    });
+
+    // TODO: tests for create web map not working
+    xdescribe('createWebMap', function() {
```

Comment:
@tomwayson was **x**describe on purpose to stop these tests from running?  I played around with these and also couldn't get the actual `expect`s to show up in the test runners either way.